tests/test_gates_gates.py

```diff
@@ -94,10 +100,12 @@ def test_sx(backend):
     # testing random expectation value due to global phase difference
     observable = random_hermitian(2**nqubits, backend=backend)
     backend.assert_allclose(
-        np.transpose(np.conj(final_state_decompose))
-        @ observable
-        @ final_state_decompose,
-        np.transpose(np.conj(target_state)) @ observable @ target_state,
+        np.transpose(np.conj(to_numpy(final_state_decompose)))
+        @ to_numpy(observable)
+        @ to_numpy(final_state_decompose),
+        np.transpose(np.conj(to_numpy(target_state)))
+        @ to_numpy(observable)
+        @ to_numpy(target_state),
```

Comment:
method called multiple times on same objects, could run one and save it in variables